python/paddle/fluid/contrib/mixed_precision/fp16_utils.py

```diff
@@ -115,6 +115,59 @@ def _insert_cast_op(block, op, idx, src_dtype, dest_dtype):
     return num_cast_ops
 
 
+def check_op_validation(ops, idx, cur_op):
+    """
+    Check whether has op (backward_role) that inputs variable in
+    cur_op's outputs behind cur_op.
+    If yes, it means _move_optimize_ops_back will cause errors
+    in program order. Therefore raise valueerror.
+    If no, return True to continue moving in _move_optimize_ops_back.
+
+    Args:
+        ops (list): A list of ops.
+        idx (int): index of cur_op in ops.
+        cur_op (Operator): Current operator.
+    """
+    OPTIMIZE = core.op_proto_and_checker_maker.OpRole.Optimize
+    for i in range(idx + 1, len(ops)):
+        op = ops[i]
```

Comment:
an alternative way is `for op in ops[idx+1:]`

Reply:
thanks!